src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -298,4 +359,177 @@ private static int timeUnitToScale(LogicalTypeAnnotation.TimeUnit timeUnit) {
             ErrorCode.INTERNAL_ERROR, String.format("Unknown time unit: %s", timeUnit));
     }
   }
+
+  /**
+   * Parses a group value based on Parquet group logical type.
+   *
+   * @param value value to parse
+   * @param type Parquet column type
+   * @param statsMap column stats map to update
+   * @param defaultTimezone default timezone to use for timestamp parsing
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @param path dot path of the column
+   * @param isDescendantsOfRepeatingGroup true if the column is a descendant of a repeating group,
+   * @return list of parsed values
+   */
+  private static ParquetBufferValue getGroupValue(
+      Object value,
+      GroupType type,
+      Map<String, RowBufferStats> statsMap,
+      ZoneId defaultTimezone,
+      final long insertRowsCurrIndex,
+      String path,
+      boolean isDescendantsOfRepeatingGroup) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return getStructValue(
+          value,
+          type,
+          statsMap,
+          defaultTimezone,
+          insertRowsCurrIndex,
+          path,
+          isDescendantsOfRepeatingGroup);
+    }
+    if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.ListLogicalTypeAnnotation) {
+      return get3LevelListValue(value, type, statsMap, defaultTimezone, insertRowsCurrIndex, path);
+    }
+    if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.MapLogicalTypeAnnotation) {
+      return get3LevelMapValue(
+          value,
+          type,
+          statsMap,
+          defaultTimezone,
+          insertRowsCurrIndex,
+          path,
+          isDescendantsOfRepeatingGroup);
+    }
+    throw new SFException(
+        ErrorCode.UNKNOWN_DATA_TYPE, logicalTypeAnnotation, type.getClass().getSimpleName());
```

Comment:
next PR: can this error have more info, with the rowindex / etc?

Reply:
Added to [jira](https://snowflakecomputing.atlassian.net/browse/SNOW-1490151).